amespahdbpythonsuite/parsers.py

```diff
@@ -0,0 +1,543 @@
+                (bool), version, comment, and number of species.
+        """
+        # Advance to root and identify namespace.
+        event, root = next(tree)
+        namespace = self.determine_xml_namespace(root)
+
+        # Root dict attributes.
+        root_dict = root.attrib
+
+        # Form important dicts, to be returned.
+        database_dict = {}
+        info_dict = {}
+        first_run = True
+
+        # Iterate through the tree, clearing elements as we go.
+        for _, (event, elem) in enumerate(tree):
+
+            # Store the top XML comment.
+            if first_run:
+                if event == "end" and elem.tag == namespace + 'comment':
+                    pahdb_comment = elem.text
+                    first_run = False
+
+            # Extract UID data on a per-tag ('specie') basis.
+            if event == "end" and elem.tag == namespace + 'specie':
+                # Extract soft data.
+                uid, metadata, comments, reference = \
+                    self._extract_info(elem, namespace)
+
+                # Extract hard data.
+                args = elem, namespace
+                geometry_dict = self._extract_numeric(*args, 'geometry')
+                transitions_dict = self._extract_numeric(*args, 'transitions')
+                laboratory_dict = self._extract_numeric(*args, 'laboratory')
+
+                # Construct the UID-specific dictionary.
+                database_dict[uid] = {
+                    'metadata': metadata,
+                    'geometry': geometry_dict,
+                    'transitions': transitions_dict,
+                    'comments': comments,
+                    'reference': reference,
+                    'laboratory': laboratory_dict,
+                }
+
+                elem.clear()
+
+        # Path of XML file for saving in dict.
+        full_path = os.path.abspath(self.filename)
+
+        # Sanity check, no repeated UIDs.
+        db_keys = [key for key in database_dict]
+        assert len(db_keys) == len(np.unique(db_keys))
+
+        # Construct dict with general XML information.
+        info_dict = {
+            'filename': full_path,
+            'type': root_dict['database'],
+            'date': root_dict['date'],
+            'full': bool(root_dict['full']),
+            'version': root_dict['version'],
+            'comment': pahdb_comment,
+            'nspecies': len(db_keys),
+        }
+
+        return database_dict, info_dict
+
+    def to_dicts(self):
+        """Return the database and info dictionaries."""
+        return self._database, self._info
+
+    @staticmethod
+    def determine_xml_namespace(root):
+        """Returns the namespace for an ElementTree root object.
+
+        Args:
+            root (ElementTree): root object.
+
+        Returns:
+            namespace (str):
+        """
+        namespace = ''
+        try:
+            namespace_end_index = root.tag.index('}')
```

Comment:
Not very elegant, but I guess this is the only way ...

Reply:
Not sure that it's easier on the eyes, but what about...
```suggestion
            namespace = root.tag.split('}')[0] + '}'
```